phases/stack.go

```diff
@@ -1,9 +1,106 @@
 package phases
 
-import "fmt"
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/go-utils/log"
+	"gopkg.in/yaml.v2"
+)
+
+var defaultStacks = map[string]string{
+	"xamarin":      "osx-vs4mac-stable",
+	"cordova":      "osx-vs4mac-stable",
+	"react-native": "osx-vs4mac-stable",
+	"ionic":        "osx-vs4mac-stable",
+	"flutter":      "osx-vs4mac-stable",
+	"android":      "linux-docker-android",
+	"macos":        "osx-xcode-10.0",
+	"ios":          "osx-xcode-10.0",
+	"other":        "",
+}
+
+var optionsStacks = []string{
+	"linux-docker-android-lts",
+	"linux-docker-android",
+	"osx-vs4mac-beta",
+	"osx-vs4mac-previous-stable",
+	"osx-vs4mac-stable",
+	"osx-xamarin-stable",
+	"osx-xcode-10.0.x",
+	"osx-xcode-10.1.x",
+	"osx-xcode-10.2.x",
+	"osx-xcode-8.3.x",
+	"osx-xcode-9.2.x",
+	"osx-xcode-9.4.x",
+	"osx-xcode-edge",
+}
+
+func getDefaultStack(bitriseYMLPath string) (string, error) {
+	data, err := ioutil.ReadFile(bitriseYMLPath)
+	if err != nil {
+		return "", fmt.Errorf("read bytrise yml (%s): %s", bitriseYMLPath, err)
+	}
+
+	var m models.BitriseDataModel
+	if err := yaml.Unmarshal(data, &m); err != nil {
+		return "", fmt.Errorf("unmarshal bitrise yml (%s): %s", bitriseYMLPath, err)
+	}
+
+	projectType := m.ProjectType
```

Comment:
what if instead of this block:
```
if m.ProjectType == "" {
  return "", nil
}
return defaultStacks[m.ProjectType], nil
```

Reply:
this way it is not required to have a key in your map with an empty value, also no need to have a new var